CameraXVideo/app/src/main/java/com/example/android/camerax/video/fragments/CameraFragment.kt

```diff
@@ -460,56 +472,57 @@ class CameraFragment : Fragment() {
-     *    which has been enumerated into:
-     *       cameraCapabilities.
-     *    User selection is saved to qualitySelectorIndex, used later in bind capture pipeline phase.
+     *    Populate a RecyclerView to display camera capabilities:
+     *       - one front facing
+     *       - one back facing
+     *    User selection is saved to qualitySelectorIndex, will be used
+     *    in the bindCaptureUsecase().
      */
     private fun initializeQualitySectionsUI() {
         val selectorStrings = cameraCapabilities[cameraIndex].qualitySelector.map {
             qualityMap.getString(it)
         }
-        // Assign adapter to ListView
-        fragmentCameraBinding.qualitySelection.adapter =
-        object : ArrayAdapter<String?>(
-            requireContext(),
-            android.R.layout.simple_list_item_1, android.R.id.text1, selectorStrings)
-        {
-            override fun getView(position: Int, tvView: View?, parent: ViewGroup): View {
-                return (super.getView(position, tvView, parent) as TextView)
-                  .apply { setTextColor(ContextCompat.getColor(requireContext(), R.color.ic_white)) }
-            }
-        }
-
-        val previousSelection = intArrayOf(-1)
-        val previousView = arrayOf<View?>(null)
-        fragmentCameraBinding.qualitySelection.let {
-            it.setOnItemClickListener { _, view, position, _ ->
-                if (previousView[0] != null) {
-                    previousView[0]!!.setBackgroundColor(0)
+        // create the adapter to QualitySelector RecyclerView
+        fragmentCameraBinding.qualitySelection.apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = GenericListAdapter(
+                selectorStrings,
+                itemLayoutId = R.layout.video_quality_item
+            ) { holderView, qcString, position ->
+
+                holderView.findViewById<TextView>(R.id.qualityTextView)?.apply {
+                    text = qcString
+                    isSelected = (position == qualitySelectorIndex)
                 }
-                view.setBackgroundColor(ContextCompat.getColor(requireContext(), R.color.icPressed))
-                previousSelection[0] = position
-                previousView[0] = view
-
-                // cache the current quality selection index
-                if (qualitySelectorIndex != position) {
+                holderView.setOnClickListener { view ->
+                    if (qualitySelectorIndex == position) return@setOnClickListener
+
+                    fragmentCameraBinding.qualitySelection.let {
+                        // deselect the previous selection on UI.
+                        it.findViewHolderForAdapterPosition(qualitySelectorIndex)
+                          ?.itemView
+                          ?.isSelected = false
+                        it.adapter?.notifyItemChanged(qualitySelectorIndex)
```

Comment:
Is this needed? The text hasn't changed and the view should already be invalidated by setting isSelected to false.

Reply:
This is needed to clear away the default selection initialized after the start up, after a new selection is made.